lib/cli/command.ts

```diff
@@ -11,45 +11,67 @@ import { XudClient, XudInitClient } from '../proto/xudrpc_grpc_pb';
  */
 export const loadXudClient = async (argv: Arguments<any>) => {
   const config = new Config();
-  await config.load({
-    xudir: argv.xudir,
-    rpc: {
-      port: argv.rpcport,
-      host: argv.rpchost,
-    },
-  });
+  try {
+    await config.load({
+      xudir: argv.xudir,
+      rpc: {
+        port: argv.rpcport,
+        host: argv.rpchost,
+      },
+    });
+  } catch (err) {
+    // if we can't load the config file, we should alert the user but continue
+    // on to attempt the call with the commands args or default config values
+    console.error(err);
+  }
 
-  const certPath = argv.tlscertpath || path.join(config.xudir, 'tls.cert');
+  // for any args that were not set, we update them to the values we
+  // determined by loading the config
+  argv.xudir = argv.xudir ?? config.xudir;
+  argv.rpcport = argv.rpcport ?? config.rpc.port;
+  if (argv.rpchost === undefined) {
+    argv.rpchost = (config.rpc.host === '0.0.0.0' || config.rpc.host === '::') ?
+      'localhost' : // if xud is listening on any address, try reaching it with localhost
+      config.rpc.host;
+  }
+
+  const certPath = argv.tlscertpath || path.join(argv.xudir, 'tls.cert');
   const cert = fs.readFileSync(certPath);
   const credentials = grpc.credentials.createSsl(cert);
 
-  // in case port and host args were not set, we update them to the values we
-  // determined by loading the config
-  argv.rpcport = config.rpc.port;
-  argv.rpchost = config.rpc.host;
-
   return new XudClient(`${argv.rpchost}:${argv.rpcport}`, credentials);
 };
 
 export const loadXudInitClient = async (argv: Arguments<any>) => {
```

Comment:
At first glance `loadXudClient` and `loadXudInitClient` look really similar to me apart from the return statement. Could we put the parsing of CLI arguments and the config into a separate method to avoid the redundant code or am I missing something?

Reply:
You're right that they're almost identical. I'd just been leaving them like that because eventually I'd like to merge the two services into one, at which point one of these methods can be thrown away. But I looked this over again and I came up with a pretty clean/simple way to refactor and remove most duplicate code, so I'm going with that.